Fixes for 47.X + Updates for 48.X + 49.X

Derpy-Jacob-903 opened the pull request:
Now uses a *Native Patch* on `Bloon.Degrade`, which is need b/c the Harmony patch crashes the game now :)
> Credit to GrahamKracker for figuring this out <3

Added Last Resort and Enchanted Glade.
> ~~Enchanted Glade uses #Ouch's image b/c the `VanillaSprites` Enum is missing that map's image.~~ Fixed!

Fix both uses of `BloonWorthMutator` missing a `bloonPropertiesToRemove` value. 
> (Rubber to Gold was changed)

Derpy-Jacob-903 committed:
Fixes for 47.X

Now uses a *Native Patch* on `Bloon.Degrade`, which is need b/c the Harmony patch crashes the game now :)
> Credit to GrahamKracker for figuring this out <3

Added Last Resort and Enchanted Glade.
> Enchanted Glade uses #Ouch's image b/c the `VanillaSprites` Enum is missing that map's image.

Fix both uses of `BloonWorthMutator` missing a `bloonPropertiesToRemove` value. (Rubber to Gold was changed)

Derpy-Jacob-903 committed:
Oops

Jacob try not to forget about the ModHelperData (challenge failed 😭)

Derpy-Jacob-903 committed:
Check for rogueGame in BossBloonManager patches

Think this should fix bosses not spawning outside of BTD6Rogue's modes (i.e. Boss Challenge)

dustinsoftware reviewed (commented):
Thanks so much! This fixed the mod on 47.

Derpy-Jacob-903 committed:
Updates for 48.X

+ Fixed Enchanted Glade's Map Image
+ Added Spa Pits Map
+ Added Spike Factory Paragon

Derpy-Jacob-903 commented:
Some Updates for 48.X!
+ Fixed Enchanted Glade's Map Image
+ Added Spa Pits Map
+ Added Spike Factory Paragon
+ Fixed Bosses not spawning in Boss Challenge/Rush and similar modes

Derpy-Jacob-903 committed:
Check for rogueGame in patches 2

Fixes Boss Challenge/Rush and Mini/Final Boss tiles in Rogue Legends

Derpy-Jacob-903 commented:
49.X Todo list:
- [ ] Fix Alch buffs, *Naval Tactics*, and *Armor Piercing Shells* (possibly others) allowing the player to purchase restricted upgrades for towers.
- [x] Add *Desperado* tower
- [x] Add the new map.
- [X] Fix *Vortex* (guess that was a one off thing?)
- [x] Fix *Dreadbloon* (Moves _**very slowly**_ with his rock armor, is this intended?)
- [X] Other bosses (*Bloonarius, Lych, Phayze, and Blasta.*) seem to work as intended.

Derpy-Jacob-903 committed:
3.2.0

Derpy-Jacob-903 committed:
3.2.1

Added Silas and support for Dirk the Dino.
> **All** modded Heroes should be supported now.
Added support for `OGDarksTowers`.
Added support for `Powers In Shop`
> Defaults to only Banana Farmers & Techbots.
Added support for Powers Pro as Towers
> Requires `Powers-In-Shop`, Defaults to disabled.
Moved `Tower` folder to `Towers`.
RoundManager drives me insane.
> don't use the restart button. it fucks shit up later.

Derpy-Jacob-903 committed:
3.2.1 hotfix

Forgot `OGDarksTowers` had paragons :clown:
also fixed a bug where `CreateTowerChoiceData` tried to make a `TowerChoice` for tier 4/5 Powers Pro (and logging failing to do so.)

Derpy-Jacob-903 committed:
3.2.2

added bomb paragon and new map
yeah

Derpy-Jacob-903 committed:
3.2.2 Hotfix

Derpy-Jacob-903 committed:
3.2.3

spooky update

Derpy-Jacob-903 committed:
3.2.4

might add Diamondback later idk

Derpy-Jacob-903 committed:
mod is fucked until someone figures out another way to get the game to set up the boss UI

mod is fucked until someone figures out another way to get the game to set up the boss UI (see `InGame_Initialise.cs`)

Derpy-Jacob-903 committed:
unfucked the boss ui

buggy but at least doesn't crash now

Derpy-Jacob-903 committed:
3.2.7p

Derpy-Jacob-903 committed:
3.2.8p

Derpy-Jacob-903 committed:
3.2.9p

Added **Skywarden** Tower.
Added **Ascent** Map.
Tower and Hero Choices no longer offer Monkeys/Heros you don't have them unlocked.
**Desperado**: `TowerAmountRanges` changed to 2 to 4 for all paths. (+1 Desperado for top path, -1 for bottom path.)
**Retribution Bloon**: Now starts spawning on round 101. (will not spawn during the 5th boss)
Tried to fix crashes after restarting during a boss (#1), it didn't work. _sigh..._
Deprecated `AllowProPowersAsMonkeys` option.
Added an `Elite` bool to `RogueBloon` (does nothing rn)
